### ISSUE TITLE
Update ShuffleLabelsOut to use sklearn.model_selection

### ISSUE THREAD

**Issue by rabitt** [marl/medleydb#71]
Currently, [medleydb.utils.artist_conditional_split](https://github.com/marl/medleydb/blob/master/medleydb/utils.py#L145) relies on the [ShuffleLabelsOut](https://github.com/marl/medleydb/blob/master/medleydb/utils.py#L200) class, which is using the soon to be deprecated sklearn.cross_validation. It needs to be updated to use sklearn.model_selection but things have changed enough that a simple switch breaks the code. @bmcfee I took a stab at this myself but don't understand what you did in the original version well enough to troubleshoot why my changes weren't working. Can you give it a go?

**Comment by bmcfee** [comment]
> @bmcfee I took a stab at this myself but don't understand what you did in the original version well enough to troubleshoot why my changes weren't working. Can you give it a go?

Sure.  FWIW, if we bump up the sklearn dependency to 0.18, we can use the built-in `GroupShuffleSplit` iterator for this and remove some code: http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GroupShuffleSplit.html#sklearn.model_selection.GroupShuffleSplit
